draft-irtf-cfrg-hpke.md

```diff
@@ -892,6 +886,39 @@ security models and assumptions, and do not consider attackers capable of quantu
 computation. A full proof of post-quantum security would need to take this
 difference into account, in addition to simply using a post-quantum KEM.
 
+## Pre-Shared Key Recommendations {#security-psk}
+
+In mode PSK and mode AuthPSK, the PSK SHOULD be of length Nh bytes or
+longer, and SHOULD have Nh bytes of entropy or more. Using a PSK shorter
+than Nh bytes is permitted. A PSK that is longer than Nh bytes or that
+has more than Nh bytes of entropy, respectively, does not increase the
+security level of HPKE, because the extraction step involving the PSK
+only outputs Nh bytes.
+
+HPKE is specified to use HKDF as key derivation function. HKDF is not
+designed to slow down dictionary attacks, see {{?RFC5869}}. Thus,
+HPKE's PSK mechanism is not suitable for use with a low-entropy
+password as the PSK: in scenarios in which the adversary knows the
+KEM shared secret zz, it can perform a dictionary attack on the PSK.
+The adversary can use different oracles to decide if a guess was good,
+depending on the scenario. First, on observation of an HPKE ciphertext,
+it can use decryption of the ciphertext as oracle. HPKE ciphertexts are
+created with an authenticated encryption scheme, which means decryption
+fails when using a wrong key. Second, the adversary can act as sender
+and create candidate HPKE ciphertexts. A recipient's behaviour can serve
+as oracle, e.g. if an error message implies that a wrong PSK was used.
+Third, if an application does not use Seal to create HPKE ciphertexts
+but only the Export interface, the adversary could rely on other
+recipient behavior which is observably different when using a wrong key.
+
+Scenarios in which the adversary can know the KEM shared secret zz
+depend on the KEM. In the case of DHKEM, it is enough if the adversary
+knows all private keys of one participant. In mode PSK, either knowledge
+of skR or skE is sufficient. If the adversary can act as sender, it does
+obviously know skE. In mode AuthPSK, either knowledge of skR or (skS and
+skE) is sufficient. Knowledge of skR is sufficient because
+Diffie-Hellman is subject to key-compromise impersonation.
```

Comment:
```suggestion
knows all private keys of one participant. 
```

I recommend deleting these examples as knowledge of all private keys of one participant is adequate.